crypto/objects/o_names.c

```diff
@@ -207,17 +209,18 @@ int OBJ_NAME_add(const char *name, int type, const char *data)
     type &= ~OBJ_NAME_ALIAS;
 
     onp = OPENSSL_malloc(sizeof(*onp));
-    if (onp == NULL) {
-        /* ERROR */
-        goto unlock;
-    }
+    if (onp == NULL)
+        return 0;
```

Comment:
This change doesn't look right?

Reply:
No, the change is right because it would unlock something it didn't have.  The lock is taken a few lines down.